statwrap/ux.py

```diff
@@ -0,0 +1,84 @@
+'''
+User Interactibility
+'''
+
+import ipywidgets as widgets
+import io
+import pandas as pd
+from IPython.display import display
+from IPython import get_ipython
+
+
+class DataUploadWidget:
+    """
+    A widget for uploading data files and creating a pandas DataFrame.
+
+    Parameters
+    ----------
+    variable_name : str
+        The name of the variable to store the DataFrame in the IPython environment.
+    accept : str, optional
+        A comma-separated string of accepted file extensions. Default is '.csv,.xls,.xlsx,.xlsm,.xlsb,.odf,.ods,.odt'.
+    auto_display : bool, optional
+        If True, the widget is displayed immediately upon creation. Default is True.
+
+    Examples
+    --------
+    Single DataFrame Usage:
+
+    >>> uploader = DataUploadWidget("df")
+
+    Multiple DataFrame Usage:
+    
+    >>> uploader1 = DataUploadWidget("df1")
+    >>> uploader2 = DataUploadWidget("df2")
+
+    Delayed Output Usage:
+
+    >>> output = widgets.Output()
+    >>> with output:
+    >>>     uploader = DataUploadWidget('df3', auto_display=False)
+    
+    >>> display(output)
+    >>> display(uploader.uploader, uploader.submit_button)
+    """
+
+    def __init__(self, variable_name, accept: str = '.csv,.xls,.xlsx,.xlsm,.xlsb,.odf,.ods,.odt', auto_display: bool = True):
+        self.accept = accept
+        self.supported = {'.csv', '.xls', '.xlsx', '.xlsm', '.xlsb', '.odf', '.ods', '.odt'}
+        self.uploader = widgets.FileUpload(accept=accept, multiple=False)
+        self.submit_button = widgets.Button(description=f"Create DF as \"{variable_name}\"")
+        self.variable_name = variable_name
+        self.auto_display = auto_display
+
+        self.uploadData()
+
+    def createDF(self):
```

Comment:
not a real blocker, but I would rename to `create_df` and rename the other method to `upload_data`, to use the pep8 convention. 

https://peps.python.org/pep-0008/#function-and-variable-names

Reply:
Which method should I rename as upload_data?

Reply:
`createDF` -> `create_df` and `uploadData` -> `upload_data`